File: app/src/main/java/com/storn/freechat/login/presenter/LoginContract.java
package com.storn.freechat.login.presenter;

import android.content.Context;
import android.support.annotation.NonNull;

import com.storn.freechat.vo.UserVo;

import java.util.List;

/**
 * Created by tianshutong on 2017/4/7.
 */

public interface LoginContract {

    interface View {

        void loadSuccess();

        void loadFailed();

        void checkFocusView();

        void showProgress(boolean active);

        void addAccountsToAutoComplete(List<?> list);

        void fillLastAccount(String userName, String password);

    }

    interface Presenter {

        void goToMainAct();

        void goToRegAct(android.view.View view, String s);

        void getAccounts();

        void saveAccount(String userName, String password);

        void checkAccount(String userName, String password);

        void attemptLogin(String userName, String password);

    }

    interface ILoginRepo {

        interface LoginTaskCallback extends Runnable {

            void onLoadSuccess();

            void onLoadFailed();
        }

        void getTask(@NonNull String uName, @NonNull String uPwd, @NonNull LoginTaskCallback callback);

        void insertOrUpdateAccount(Context context, UserVo userVo);
    }
}

File: app/src/main/java/com/storn/freechat/login/presenter/LoginContract.kt
package com.storn.freechat.login.presenter

import android.content.Context

import com.storn.freechat.vo.UserVo

/**
 * 登录接口
 * Created by tianshutong on 2017/4/7.
 */

interface LoginContract {

    interface View {

        fun loadSuccess()

        fun loadFailed()

        fun checkFocusView()

        fun showProgress(active: Boolean)

        fun addAccountsToAutoComplete(list: List<String>)

        fun fillLastAccount(userName: String, password: String)

    }

    interface Presenter {

        fun goToMainAct()

        fun goToRegAct(view: android.view.View, s: String)

        fun getAccounts()

        fun saveAccount(userName: String, password: String)

        fun checkAccount(userName: String, password: String)

        fun attemptLogin(userName: String, password: String)

    }

    interface ILoginRepo {

        fun insertOrUpdateAccount(context: Context, userVo: UserVo)
    }

    interface ILoginListener {

        fun start()

        fun success()

        fun fail(error: String)

    }

    interface IChangePwdListener {
        fun start()

        fun success()

        fun fail(error: String)
    }
}
